var/spack/repos/builtin/packages/rocm-opencl/package.py

```diff
@@ -194,14 +194,15 @@ def url_for_version(self, version):
 
     @classmethod
     def determine_version(cls, lib):
-        match = re.search(r"lib\S*\.so\.\d+\.\d+\.(\d)(\d\d)(\d\d)", lib)
+        detected_version = None
+        with open(lib, "rb") as f:
+            library_content = str(f.readlines())
```

Comment:
Are we really scanning the content of a library? Is that necessary for detection?

Reply:
Unfortunately I think scanning the contents of ```libOpenCL.so``` file is the only way to determine the ROCm version from an arbitrary external installation. From what I've gathered there's no other reliable way to determine that the detected OpenCL library was actually built for ROCm, and if so, what version of ROCm:
- ```clinfo``` that ships with ROCm only prints info about the AMD GPUs available on the system.
- All of the include files don't seem to have any Macros indicating HIP/ROCm versions
- The folder naming convention for ROCm is the last way for detecting the ROCm version, which can be arbitrarily changed by a user that builds this package without Spack
The library file has the ROCm version embedded. The advantage is that it doesn't confuse it with other OpenCL vendors and can detect the correct version of ROCm it was built for.

Scanning the content of the library doesn't seem worse than regex matching the version from the filename.

Reply:
@alalazo @renjithravindrankannath @srekolam any updates on this? If there's a better way to detect ROCm OpenCL versions I'd be more than happy to modify this PR.

Reply:
@amd-toolchain-support Do we have an answer to this :point_up: ?